### PR TITLE
[DASH] Add ACL tags stage 1 HLD.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**

### CODE REVIEW

File: doc/dash-acl-tags/dash-acl-tags.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+# SAI API
+
+Stage 1 implementation does not require SAI API.
+
+# Configuration and management
+
+## CLI/YANG model Enhancements
+
+No CLI/YANG enhancements required.
```
Comment (dgsudharsan) [inline doc/dash-acl-tags/dash-acl-tags.md:301]: Aren't we exposing DASH_PREFIX_TAG_TABLE to config_db? In that case we should update yang model

File: doc/dash-acl-tags/dash-acl-tags.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+- Provide support for adding new ACL tags with the list of prefixes in the DASH_PREFIX_TAG_TABLE APP DB table.
+- Provide support for updating list of prefixes for ACL tags. 
+- Provide support for removing ACL tags. ACL tag can be removed only if it is not attached to any ACL rules.
+- Provide support for multiple source and destination ACL tags for ACL rules in DASH_ACL_RULE_TABLE.
+- Provide support for support for TAGs and list prefixes configuration for ACL rules in DASH_ACL_RULE_TABLE.
+
+### Configuration flow
+
+![Configuration flow](images/dash-acl-tags.png)
```
Comment (dgsudharsan) [inline doc/dash-acl-tags/dash-acl-tags.md:165]: Isn't ACL configured from config_db? I see DASH_ACL_RULE and DASH_ACL_GROUP in config_db schema

File: doc/dash-acl-tags/dash-acl-tags.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+No CLI/YANG enhancements required.
+
+## Config DB Enhancements
+
+No Config DB Enhancements required
+
+# Warmboot and Fastboot Design Impact
+DPU SONiC doesn't support warm/fast boot
+
+# Restrictions/Limitations
```
Comment (dgsudharsan) [inline doc/dash-acl-tags/dash-acl-tags.md:310]: Should we mention how we handle tags with overlapping prefixes? I prefer we capture the handling in some section and update the limitations if any here

File: doc/dash-acl-tags/dash-acl-tags.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+| Create ACL rule with multiple source and multiple destination tags | Verify that ACL rule with correct attributes is created |
+| Create ACL rule with multiple source and multiple destination tags and list of source and destination prefixes | Verify that ACL rule with correct attributes is created |
+| Update ACL tag used by ACL rule from ACL group not bound to ENI | Verify that ACL rule with is updated with correct attributes |
+|Update ACL tag used by ACL rule from ACL group bound to the ENI | Verify the following <br />- new ACL group is created <br /> - all ACL rules are copied to the new group <br /> - appropriate rule is updated <br /> - new ACL group is bound to the ENI <br /> - old ACL rules are removed <br /> - old ACL group is removed |
+|Create tag with missing mandatory fields|Verify that error message is sent to syslog|
+|Create tag with incorrect data | Verify that error message is sent to syslog|
+|Create ACL rule with not existing tag |Verify that rule is not created|
+|Create ACL rule with not existing tag than create a tag |Verify that rule is created after the tag is added|
+
```
Comment (dgsudharsan) [inline doc/dash-acl-tags/dash-acl-tags.md:331]: Can we add a test case to confirm delayed creation logic? First lets create a rule with non existing tag and then add the tag later.
Similarly lets remove a tag first and then remove the associated rule later. We need to verify both tag and rule is deleted in this scenario.

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: doc/dash-acl-tags/dash-acl-tags.md (hunk 1 above)
Comment (oleksandrivantsiv) [reply doc/dash-acl-tags/dash-acl-tags.md:301]: All DASH configuration is in APP DB.

File: doc/dash-acl-tags/dash-acl-tags.md (hunk 2 above)
Comment (oleksandrivantsiv) [reply doc/dash-acl-tags/dash-acl-tags.md:165]: The schema is for the APP DB. Microsoft had a plan to add configuration validation for APP DB as well.
Right now the schema is outdated. I checked with Prince, and currently, we don't need to update it.

File: doc/dash-acl-tags/dash-acl-tags.md (hunk 3 above)
Comment (oleksandrivantsiv) [reply doc/dash-acl-tags/dash-acl-tags.md:310]: Yes, but this is a stage 2 feature. For stage 1 it is not required.

File: doc/dash-acl-tags/dash-acl-tags.md (hunk 4 above)
Comment (oleksandrivantsiv) [reply doc/dash-acl-tags/dash-acl-tags.md:331]: Sure, I will update the test cases

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**